src/gui/sharemanager.h

```diff
@@ -237,23 +246,38 @@ class LinkShare : public Share
      * In case of a server error the serverError signal is emitted.
      */
     void setExpireDate(const QDate &expireDate);
-
+    
+    /*
+     * Set the label of the share link.
+     */
+    void setLabel(const QString &label);
+    
+    /*
+     * Create OcsShareJob and connect to signal/slots
+     */
+    template <typename LinkShareSlot>
```

Comment:
Not sure if it makes sense to prefix the template parameter with `T` to make clear that it is not an actual type?